agent/pkg/controllers/config_controller.go

```diff
@@ -28,6 +28,15 @@ func PostTapConfig(c *gin.Context) {
 		return
 	}
 
+	if kubernetesProvider == nil {
+		var err error
+		kubernetesProvider, err = kubernetes.NewProviderInCluster()
```

Comment:
can we use here a singleton pattern? I am sure there are other places in code we use this provider

Reply:
your current implementation is not a thread safe ..

Reply:
@RoyUP9 i will merge this PR because of the time pressure, but i think we need to implement a singleton here.